geocodingv7/geocoding_test.go

```diff
@@ -0,0 +1,105 @@
+package geocodingv7_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"testing"
+
+	"go.einride.tech/here/geocodingv7"
+	"gotest.tools/v3/assert"
+)
+
+type GeocodingMock struct {
+	responseStatus int
+	responseBody   geocodingv7.GeocodingResponse
+}
+
+func (c *GeocodingMock) Do(req *http.Request) (*http.Response, error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	b, err := json.Marshal(c.responseBody)
+	if err != nil {
+		return nil, err
+	}
+	r := bytes.NewReader(b)
+	return &http.Response{
+		StatusCode:    c.responseStatus,
+		Header:        headers,
+		Body:          io.NopCloser(r),
+		ContentLength: int64(len(b)),
+	}, nil
+}
+
+func TestGeocodingService_Geocoding(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run(
+		"when neither address or geo position is given as input, then return invalid argument",
```

Comment:
Is this one still relevant if geocoding and reverse geocoding are separate endpoints? Don't we always need an address now?

Reply:
test is still needed but I changed the comment. We want to check that either Query or Address is given